pykern/pkcli/ci.py

```diff
@@ -73,6 +66,50 @@ def run():
     from pykern.pkcli import fmt, test
     from pykern import pkio
 
+    check_eof_newline()
     check_prints()
     fmt.diff(pkio.py_path())
     test.default_command()
+
+
+def _check_files(case, check_file):
+    def _error(m):
+        pkcli.command_error("{}: {}", case, m)
+
+    d = PKDict(
+        check_eof_newline=PKDict(
+            exclude_files=re.compile(
+                r"/static/js/ext/|node_modules/|^run/|tests/|^venv/"
+            ),
+            include_files=re.compile(r"\.(html|jinja|js|json|md|py|tsx|yml)$"),
+        ),
+        check_prints=PKDict(
+            exclude_files=re.compile(
+                f".*(?:{pkunit.DATA_DIR_SUFFIX}|{pkunit.WORK_DIR_SUFFIX})/"
+                + f"|^\\w+/{pksetup.PACKAGE_DATA}/"
+                + r"|pkdebug[^/]*\.py$"
+                + r"|(?:^|/)\."
+                + r"|^run/"
+                + r"|^venv/"
+            ),
+            include_files=re.compile(r".py$"),
+        ),
+    )[case]
+
+    r = []
+    p = pkio.py_path()
+    n = 0
+    for f in pkio.walk_tree(p, d.include_files):
+        f = p.bestrelpath(f)
+        if re.search(d.exclude_files, f):
+            continue
+        n += 1
+        c = check_file(pkio.read_text(f).split("\n"))
+        if c is not None:
```

Comment:
I see that None was not the best choice. Rather, return an empty list if there are no errors. Then the code would be simpler. You could just say `for l in check_file(...)` and not need the temporary.